src/Indicator.vala

```diff
@@ -210,6 +218,23 @@ public class Notifications.Indicator : Wingpanel.Indicator {
             warning ("Failed to open notifications settings: %s", e.message);
         }
     }
+
+    private void update_tooltip () {
+        uint number_of_notifications = Session.get_instance ().get_session_notifications ().length ();
+
+        switch (number_of_notifications) {
+            case 0:
+                dynamic_icon.tooltip_markup = Granite.markup_accel_tooltip ({}, _("No notifications"));
+                break;
+            case 1:
+                dynamic_icon.tooltip_markup = Granite.markup_accel_tooltip ({}, _("1 notification"));
+                break;
+            default:
+                /* Anything else */
+                dynamic_icon.tooltip_markup = Granite.markup_accel_tooltip ({}, _("%u notifications from %i apps".printf (number_of_notifications, nlist.app_entries.size)));
```

Comment:
We might need to be more robust with how we create this string; technically you could have 3 notifications from one app, and this would incorrectly pluralize "app".

Reply:
Ahh didn't think about that. I'll work on that.

Reply:
You'll want to use ngettext here since plurals work differently in some languages than in English